### PR TITLE
Add endpoint for listing letters pending upload

### PR DESCRIPTION
**Pull request by rkondratowicz**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/PendingLettersService.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class PendingLettersService {
+
+    private final LetterRepository repo;
+
+    public PendingLettersService(LetterRepository repo) {
+        this.repo = repo;
+    }
+
+    public List<Letter> getPendingLetters() {
+        return repo
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/PendingLettersService.java]: Based on this code I assume this repository method returns a stream. If that's the case, that stream has to be closed in every case (success or failure), so we need to wrap it in try-with-resources. There's a DB connection and transaction related with it.

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/PendingLettersService.java]: Also, there will have to be a `@Transactional` annotation somewhere, to make it work.

File: src/test/java/uk/gov/hmcts/reform/sendletter/services/PendingLettersServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import uk.gov.hmcts.reform.sendletter.entity.LetterStatus;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+class PendingLettersServiceTest {
+
+    @Test
+    void should_read_letters_in_proper_status_from_repo() {
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/sendletter/services/PendingLettersServiceTest.java]: We also need to verify the result of the method.

File: src/test/java/uk/gov/hmcts/reform/sendletter/services/PendingLettersServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        // given
+        LetterRepository repo = mock(LetterRepository.class);
+        PendingLettersService service = new PendingLettersService(repo);
+        List<Letter> noLetters = Collections.emptyList();
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/sendletter/services/PendingLettersServiceTest.java]: Empty list is a good case to test 👍. It would also be useful to have a test for a non-empty list. Alternatively, if we just care if the service has returned what it got the repository, we can create an assertion for exactly the same instance.

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/sendletter/services/PendingLettersServiceTest.java]: did not want to tamper too much with existing pr but can include extra test just for the sake of validity
